packages/flutter_tools/lib/src/windows/visual_studio.dart

```diff
@@ -416,17 +420,40 @@ class VswhereDetails {
 
     return VswhereDetails(
       meetsRequirements: meetsRequirements,
-      installationPath: details['installationPath'] as String?,
-      displayName: details['displayName'] as String?,
-      fullVersion: details['installationVersion'] as String?,
       isComplete: details['isComplete'] as bool?,
       isLaunchable: details['isLaunchable'] as bool?,
       isRebootRequired: details['isRebootRequired'] as bool?,
       isPrerelease: details['isPrerelease'] as bool?,
+
+      // Below are strings that must be well-formed without replacement characters.
+      installationPath: _validateString(details['installationPath'] as String?),
+      displayName: _validateString(details['displayName'] as String?),
```

Comment:
I had meant my comments to apply to `displayName` as well. Why do we need this to be well-formed?

Reply:
My mistake, fixed